File: Practice/20/C++/20/20/20.cpp
#include <iostream>
#include <string>
using namespace std;
int main()
{
	int n,k, in;
	cin >> n >> k;
	double min = n + 1, cmax = 0;
	string *a = new string[k];
	int *b = new int[k];
	int *c = new int[k];
	for (int i = 0; i < k; i++) {
		cin >> a[i] >> b[i] >> c[i];
		if (b[i] / c[i] <= min && c[i] > cmax) { min = b[i]; in = i; }
	}
	if (min == n + 1) cout << -1;
	else cout << a[in] << " " << n / b[in] << endl << n / b[in] * c[in] << endl << n - n / b[in] * b[in] << endl;
}

File: Practice/20/Python/20.py
n = int(input())
k = int(input())
mini = 1.1
mini = n + 1
cmax = 0
ink = 0
a = []
b = []
c = []
for i in range (k):
  a.append(input())
  b.append(int(input()))
  c.append(int(input()))
  if b[i]/c[i]<=mini and c[i] > cmax:
    mini=b[i]
    ink=1
if min == n + 1: print(-1)
else: print(a[ink],' ', n/b[ink], '\n', n/b[ink]*c[i], '\n', n-n/b[ink]*b[ink],'\n',sep='')